src/libServer/LookupServer.cpp

```diff
@@ -1034,52 +1040,77 @@ Json::Value LookupServer::GetEthBlockByNumber(const std::string& blockNumberStr,
     if (txBlock == NON_EXISTING_TX_BLOCK) {
       return Json::nullValue;
     }
-    const auto dsBlock =
-        m_mediator.m_dsBlockChain.GetBlock(txBlock.GetHeader().GetDSBlockNum());
-
-    std::vector<TxBodySharedPtr> transactions;
-    std::vector<TxnHash> transactionHashes;
+    return GetEthBlockCommon(txBlock, includeFullTransactions);
 
-    // Gather either transaction hashes or full transactions
-    const auto& microBlockInfos = txBlock.GetMicroBlockInfos();
-    for (auto const& mbInfo : microBlockInfos) {
-      if (mbInfo.m_txnRootHash == TxnHash{}) {
-        continue;
-      }
-
-      MicroBlockSharedPtr microBlockPtr;
-      if (!BlockStorage::GetBlockStorage().GetMicroBlock(
-              mbInfo.m_microBlockHash, microBlockPtr)) {
-        continue;
-      }
+  } catch (std::exception& e) {
+    LOG_GENERAL(INFO, "[Error]" << e.what() << " Input: " << blockNumberStr
+                                << ", includeFullTransactions: "
+                                << includeFullTransactions);
+    throw JsonRpcException(RPC_MISC_ERROR, "Unable To Process");
+  }
+}
 
-      const auto& currTranHashes = microBlockPtr->GetTranHashes();
-      if (!includeFullTransactions) {
-        transactionHashes.insert(transactionHashes.end(),
-                                 currTranHashes.begin(), currTranHashes.end());
-        continue;
-      }
-      for (const auto& transactionHash : currTranHashes) {
-        TxBodySharedPtr transactioBodyPtr;
-        if (!BlockStorage::GetBlockStorage().GetTxBody(transactionHash,
-                                                       transactioBodyPtr)) {
-          continue;
-        }
-        transactions.push_back(std::move(transactioBodyPtr));
-      }
+Json::Value LookupServer::GetEthBlockByHash(const std::string& inputHash,
+                                            bool includeFullTransactions) {
+  try {
+    const BlockHash blockHash{inputHash};
+    const auto txBlock = m_mediator.m_txBlockChain.GetBlockByHash(blockHash);
+    static const TxBlock NON_EXISTING_TX_BLOCK{};
```

Comment:
Is it neccessary to use a static in this way? I consider it to be a little unsafe to have static classes as it means whatever is in its constructor will execute at t = 0. In this case it is fine as there is no constructor but this has caused issues for me in the past where you can get race conditions at start up.

Could be something like:

```
    if (!m_mediator.m_txBlockChain.GetBlockByHash(blockHash, setMe)) {
      return Json::nullValue;
    }
```

Reply:
Are we going to rely on automatically generated `operator bool()`? I'm afraid this may not work (some of the fields may not return false on an empty object). I think we can just use my approach without `static` though.